apartmenttherapy/react-gpt
# React v17 support

#### What does this PR do?

Bumps dependencies to support React 17. This is package blocks some of our other apps and services from being able to upgrade due to peerDependencies (namely oxygen and the auth migration).

#### How can this change be manually tested?

I ran the tests with `yarn test`. It appears two tests are failing. But they appear to be the same two tests that fail on `master`. Not sure how you guys are rolling with this. Are the tests known failures? 

>SUMMARY:
>✔ 65 tests completed
>✖ 2 tests failed
>
>FAILED TESTS:
 > Bling
 >   ✖ reflects outOfPage props to adSlot
 >     Chrome 110.0.0 (Mac OS X 10.15.7)
 >   Error: Uncaught TypeError: Cannot read properties of undefined (reading 'INTERSTITIAL') (webpack:///src/Bling.js:9:23639 <- test/Bling.spec.js:21225)
>
>    ✖ sets slotSize to 0,0 on foldCheck of 'fluid' or ['fluid']
>      Chrome 110.0.0 (Mac OS X 10.15.7)
>    Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

To test the ad loading stuff:
1. pull down this branch. 
2. In the `react-gpt` directory, `yarn` and `yarn build` then, run `npm link`
3. Open up oxygen in your terminal
4. Switch to the `s/react17` branch
5. Run `npm link @atmedia/react-gpt`
6. Load up the sites as you normally would locally.

#### Asana Card

https://app.asana.com/0/1201709980437746/1203717210397677/f

#### Questions or concerns about this change?

~When I bump the React dependencies to v17, I get a peerDependency warning for [Radium](https://www.npmjs.com/package/radium). This is a bummer because it looks like it's a package that is no longer [supported by the maintainers.](https://github.com/formidablelabs/radium) They encourage forks, so... I guess we do that?~

#### Deployment Notes

~Need to resolve the peerDependency on Radium and React v16~ Radium has React 17 support in 0.26.2

---

#### PR Readiness

* [ ] Changes have been validated in local dev

#### Code Readiness

* [ ] Code follows [AT Media styles and conventions](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/RCwBYjgA4B)
* [ ] Code has been [auto-formatted and is free of lint errors](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/BXnBJVigtW)
* [ ] Unrelated code changes are not present
* [ ] Relevant inline documentation or READMEs have been updated
* [ ] Application and browser logs reviewed for runtime warnings and errors
* [ ] Correct level of [logging](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/gwTZjrJhOy) added to code

#### Automated Testing

* [ ] [Unit tests](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/k6Uah8Iquw) are written at the appropriate level given the code’s risk-level
* [ ] No unnecessary test runner output, e.g. debugging or warnings are not present in Jest or Unittest output

#### Other Considerations

* [ ] Relevant documentation updated in Slite

Also consider this PR's impact on architecture and security; AMP, Apple News, RSS; SEO, GA, performance and accessibility.

Please see [Post Merge](https://apartmenttherapy.slite.com/app/docs/88ULglsAB9) for additional responsibilities after this PR is merged.
